### PR TITLE
788 - Add missing Cypress tests to achieve better coverage

### LINKED ISSUE #788: Add missing Cypress tests to achieve full coverage

### PR DESCRIPTION
**Pull request by narendrasss**
Closes #788 

This PR improves code coverage by around ~18% from 43% to 61% statement coverage. It also refactors some of the tests to better adhere to best practices. Specifically:

- Minimize using the node modal to make changes to the node
  - Using the node modal takes time and leads to needless repetition. This new method means we test the modal happy path once, and other tests programmatically make updates to the nodes (through Vuex actions).
- Removes the `applyModalChanges` utility function so it's more obvious what we're actually testing

### Other Notes

- The test runs are approaching 10 minutes per run. I think we should look into integrating the [Cypress Dashboard](https://dashboard.cypress.io) and implementing [parallelization](https://github.com/cypress-io/github-action#parallel) to cut this time down.
- What kind of coverage are we looking to achieve? 
- I was using the "Given-When-Then" methodology to name tests, but I quickly found that it became tedious to create clear test names - especially since in E2E tests there are a lot of different assertions and events happening in one run. In one of the spec files, `settings.spec.js`, I used the normal convention but I didn't change the naming for the other tests that were already written.

### CODE REVIEW

**Comment by narendrasss** [comment]
Look into programmatically adding H5P content.

File: templates/vue/src/utils/Helpers.js
Comment (wynnset) [inline templates/vue/src/utils/Helpers.js:145]: Wouldn't it be better if this is also calling `Helpers.deepMerge`?

**Review by wynnset — CHANGES_REQUESTED** [review]
Thank you for writing all these tests - they look really good! I skimmed this a bit as I think I'll need to look at it in more detail later, but I found just a couple of minor things you might want to look at, but otherwise this is good to get merged after we merge the base branch into master first. You can mark it as ready to merge once you have double-check the items below.

File: templates/vue/cypress/integration/settings.spec.js
Comment (narendrasss) [reply templates/vue/cypress/integration/settings.spec.js:10]: Yes, I mentioned this in the description:

> I was using the "Given-When-Then" methodology to name tests, but I quickly found that it became tedious to create clear test names - especially since in E2E tests there are a lot of different assertions and events happening in one run. In one of the spec files, `settings.spec.js`, I used the normal convention but I didn't change the naming for the other tests that were already written.

What are your thoughts?

Comment (wynnset) [reply templates/vue/cypress/integration/settings.spec.js:10]: Oh sorry I didn't see that. I think it would make sense to use the same naming convention, so I suggest either changing the old names or keep following Given-When-Then. That said, if you think it takes too long and it doesn't matter for some reason, I trust your judgement.

**Comment by wynnset** [comment]
@narendrasss please fix merge conflicts and reassign back to me.

File: templates/vue/cypress/integration/link-authoring.spec.js
Comment (wynnset) [inline templates/vue/cypress/integration/link-authoring.spec.js:4]: Could we name this something less generic than `stump` in case in the future we want to add other types of tapestries? Something like `base-tapestry` or something?

File: templates/vue/cypress/integration/node-appearance.spec.js
Comment (wynnset) [inline templates/vue/cypress/integration/node-appearance.spec.js:19]: Could we use a creative commons image (e.g. Wikimedia) and make sure the URL is reliable enough that it won't break in the future?

**Review by wynnset — COMMENTED** [review]
I can make these updates, but just wanted to first get your feedback about it @narendrasss

File: templates/vue/cypress/integration/link-authoring.spec.js
Comment (narendrasss) [reply templates/vue/cypress/integration/link-authoring.spec.js:4]: There's already other types of tapestries under `fixtures`, I think this `stump` name refers to the fact it's a root node with two nodes so it's like a  "mini-tree" 😄

Feel free to rename it to something more appropriate though.

File: templates/vue/cypress/integration/node-appearance.spec.js
Comment (narendrasss) [reply templates/vue/cypress/integration/node-appearance.spec.js:19]: Yes that sounds good!

**Pull request merged**
